tsdb/chunks/head_chunks.go

```diff
@@ -870,6 +870,12 @@ func (cdm *ChunkDiskMapper) Truncate(mint int64) error {
 		cdm.CutNewFile()
 	}
 	errs.Add(cdm.deleteFiles(removedFiles))
+
+	if cdm.curFileSequence == 0 && len(chkFileIndices) == len(removedFiles) {
```

Comment:
Is there any possible case where `cdm.curFileSequence != 0 && len(chkFileIndices) == len(removedFiles)`?
I'm just trying to understand why the `cdm.curFileSequence != 0` is necessary, it seems like a safety check to be really really sure that we want to reset the `evtlPos()` to `0`, right?

Reply:
also, doesn't reading `cmd.curFileSequence` require `cdm.readPathMtx` to be held?

Reply:
`cdm.curFileSequence` is 0 until a new file has not been created after a restart. It is never 0 once a new file is cut, where it starts from 1. And when there is an active file, the current file is never truncated. So we only run into the case of the evtlPos resetting only on startup when no new file is cut yet.

And you are right about the lock, will update that, thanks!

Reply:
I will add a comment as well to clarify why it is only checked for `==0`

Reply:
Thinking about it again, we should just check `len(chkFileIndices) == len(removedFiles)` to avoid any future regressions, since the file creation happens from disk and can prevent any regression.